### PR TITLE
chore: libwaku - extending the library with peer_manager and peer_exchange features

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
One more step towards exposing nwaku through libwaku. The exposed features are currently needed by status-go.
Notice that there will be more upcoming features soon.

# Changes

Adding new features:
- waku_get_peerids_from_peerstore
- waku_get_peerids_by_protocol
- waku_peer_exchange_request

### CODE REVIEW

File: examples/cbindings/waku_example.c [hunk 1]
```diff
@@ -326,6 +326,10 @@ int main(int argc, char** argv) {
                                   event_handler,
                                   userData) );
 
+    WAKU_CALL( waku_get_peerids_from_peerstore(ctx,
+                                  event_handler,
+                                  userData) );
```
Comment (gabrielmer) [inline examples/cbindings/waku_example.c:331]: Why do we need this call?

File: library/libwaku.nim [hunk 2]
```diff
@@ -658,5 +698,21 @@ proc waku_stop_discv5(
...
+  let discoveredPeers = waku_thread.sendRequestToWakuThread(
+    ctx, RequestType.DISCOVERY, DiscoveryRequest.createPeerExchangeRequest(numPeers)
+  ).valueOr:
+    let msg = $error
+    callback(RET_ERR, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+    return RET_ERR
+
+  let msg = $discoveredPeers
+  callback(RET_OK, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+  return RET_OK
```
Comment (SionoiS) [inline library/libwaku.nim:715]: Can't this repeated part be abstracted?

File: library/waku_thread/inter_thread_communication/requests/peer_manager_request.nim [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+import
+  ../../../../waku/factory/waku,
+  ../../../../waku/node/waku_node,
+  ../../../alloc,
+  ../../../../waku/node/peer_manager
 
-type PeerManagementMsgType* = enum
+type PeerManagementMsgType* {.pure.} = enum
   CONNECT_TO
+  GET_PEER_IDS_FROM_PEER_STORE
```
Comment (SionoiS) [inline library/waku_thread/inter_thread_communication/requests/peer_manager_request.nim:11]: Maybe `GET_ALL_PEER_IDS` ?

**Review by SionoiS — APPROVED** [review]
LGTM Thank you!

Left some comments.

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM!!!!!

File: library/libwaku.nim (hunk 2 above)
Comment (Ivansete-status) [reply library/libwaku.nim:715]: Yup, I will double-check that in a separate PR

File: library/waku_thread/inter_thread_communication/requests/peer_manager_request.nim (hunk 3 above)
Comment (Ivansete-status) [reply library/waku_thread/inter_thread_communication/requests/peer_manager_request.nim]: Much better thanks! Added in commit: https://github.com/waku-org/nwaku/pull/3026/commits/9268abbb1ddc006dc909bc8c04ede3b7054a61fd

**Pull request merged**